README.md

```diff
@@ -135,15 +140,62 @@ npm install -D @types/vscode
 
 ### Usage
 
-Just import it as if you were in a vscode extension:
+You can just import it as if you were in a vscode extension:
 
 ```typescript
 import * as vscode from 'vscode'
+import { initialize } from 'vscode/extensions'
+
+await initialize()
 
 const range = new vscode.Range(...)
 vscode.languages.registerCompletionItemProvider(...)
 ```
 
+The api will use the manifest of a default vscode extension, which can be overriden by providing it to the `initialize` function.
+
+You can also register a new extension from its manifest:
+```typescript
+import { registerExtension, initialize as initializeVscodeExtensions } from 'vscode/extensions'
+
+await initialize()
+
+const { registerFile: registerExtensionFile, api: vscodeApi } = registerExtension(defaultThemesExtensions)
+
+registerExtensionFile('/file.json', async () => fileContent)
+vscodeApi.languages.registerCompletionItemProvider(...)
+```
+
+### Default vscode extensions
+
+VSCode uses a bunch of default extensions. Most of them are used to load the default languages and grammars (see https://github.com/microsoft/vscode/tree/main/extensions).
+
+This library bundles most of them and allows to import the ones you want:
+```typescript
+import 'vscode/default-extensions/javascript'
+import 'vscode/default-extensions/json'
+...
+```
+
+### Loading vsix file
+
+VSCode extension are bundled as vsix files.
+This library exposes a rollup plugin (vite-compatible) that allows to load a vsix file. The code is not used, only the declarative part in the manifest.
+
+- rollup/vite config:
+```typescript
+import vsixPlugin from 'vscode/rollup-vsix-plugin'
```

Comment:
Have you already made use of it somewhere?

Reply:
I did on the demo with the ocaml vscode extension. The demo already contains a lot of stuff and I don't really know how to showcase it properly